api4/post.go

```diff
@@ -185,6 +194,17 @@ func getPostsForChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		mlog.Error("Failed to prepare posts for getPostsForChannel response", mlog.Any("err", err))
 	}
 
+	if nextPostId == "" {
```

Comment:
Similarly, we might want to differentiate here between the case that we know the next/previous post IDs are empty (ie you're at the start/end of the channel) so that we aren't doing this when we don't need to

Reply:
As commented above, I only applied that to `previousPostId` but not to `nextPostId`.